src/testing/AddTestPage.test.tsx

```diff
@@ -0,0 +1,234 @@
+import React from 'react';
+import { Router, Route } from 'react-router-dom';
+import { createMemoryHistory, History } from 'history';
+import { wait, render, screen, fireEvent } from '@testing-library/react';
+
+import { AddTestPage } from './AddTestPage';
+
+import { fetchUser, fetchTestTypes, User, Sex, TestType, createTest, fetchCountries } from '../api';
+import { useAuthentication } from '../authentication';
+
+jest.mock('../api');
+jest.mock('../authentication');
+
+const fetchUserMock = fetchUser as jest.MockedFunction<typeof fetchUser>;
+const fetchTestTypesMock = fetchTestTypes as jest.MockedFunction<typeof fetchTestTypes>;
+const fetchCountriesMock = fetchCountries as jest.MockedFunction<typeof fetchCountries>;
+const createTestMock = createTest as jest.MockedFunction<typeof createTest>;
+const useAuthenticationMock = useAuthentication as jest.MockedFunction<typeof useAuthentication>;
+
+describe('Test adding page', () => {
+  let history: History;
+  let debug;
+
+  beforeEach(() => {
+    history = createMemoryHistory();
+    fetchUserMock.mockImplementation(() => Promise.resolve(aUser()));
+    useAuthenticationMock.mockImplementation(() => ({
+      token: 'mock-token',
+      userId: 'mock-user',
+      authenticate: jest.fn(),
+      signOut: jest.fn(),
+      hasPermission: (key: string) => key === 'mock-permission',
+    }));
+    fetchTestTypesMock.mockImplementation(() =>
+      Promise.resolve([aSimpleTestType(), aTestType(), aNonPermittedTestType()]),
+    );
+    fetchCountriesMock.mockImplementation(() =>
+      Promise.resolve([
+        { code: 'GB', name: 'United Kingdom' },
+        { code: 'EE', name: 'Estonia' },
+      ]),
+    );
+
+    debug = render(
+      <Router history={history}>
+        <Route path="/users/:userId/add-test">
+          <AddTestPage />
+        </Route>
+      </Router>,
+    ).debug;
+  });
+
+  describe('when testing yourself', () => {
+    it('lets you add a test result', async () => {
+      history.push('/users/mock-user/add-test');
+      await wait(() => expect(screen.queryByText(/test type/i)).toBeTruthy());
+      expect(createTestMock).not.toHaveBeenCalled();
```

Comment:
Would be tempted to move the above lines into their own expectation, just as this test is massive and they don't directly relate to the described situation.